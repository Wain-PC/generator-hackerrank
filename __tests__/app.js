"use strict";
const path = require("path");
const assert = require("yeoman-assert");
const helpers = require("yeoman-test");
const {readFile} = require('fs').promises;

describe("generator-hackerrank:app", () => {
  const opts = {
    topic: 'Sample Topic',
    task: "Sample Task",
    inputVarName: "n",
    unitTestsNumber: 1,
    exampleInput1: "testInput",
    exampleOutput1: "testOutput"
  };

  const taskFilePath = `${opts.topic}/${opts.task}.js`;
  const testFilePath = `${opts.topic}/${opts.task}.test.js`;

  beforeAll(() => {
    return helpers.run(path.join(__dirname, "../generators/app")).withPrompts(opts);
  });

  it("creates files", () => {
    assert.file(taskFilePath);
    assert.file(testFilePath);
  });

  it("task file content is correct", async () => {
    const contents = await readFile(taskFilePath, {encoding: "utf8"});
    expect(contents).toMatchSnapshot();
  });

  it("test file content is correct", async () => {
    const contents = await readFile(testFilePath, {encoding: "utf8"});
    expect(contents).toMatchSnapshot();
  });
});
