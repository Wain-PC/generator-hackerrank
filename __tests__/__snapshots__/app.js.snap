// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generator-hackerrank:app task file content is correct 1`] = `
"const solve = (n) => {
  // Put your solution here
  return n;
};

module.exports = { solve };
"
`;

exports[`generator-hackerrank:app test file content is correct 1`] = `
"const { solve } = require('./Sample Task');

describe('Sample Task', () => {
  const examples = [

    {
      query: [1, 2, 3]
      answer: 5,
    },

  ];

  examples.forEach(({ query, answer }, i) => {
    it(\`should solve example \${i}\`, () => {
      expect(solve(query)).toEqual(answer);
    });
  });
});
"
`;
